functions/CheckForCustomer.js

```diff
@@ -1,148 +1,217 @@
+        .then(buildResponse)
+        .catch(handleError)
+        .then(() => callback(out))
+        .catch(error => {
+            logError(`The callback function threw an exception: ${error}`);
+            setTimeout(() => {
+                throw error;
+            });
+        });
+
+    function logInfo(msg) {
+        nc.log(msg, "info");
+    }
+
+    function logWarn(msg) {
+        nc.log(msg, "warn");
+    }
+
+    function logError(msg) {
+        nc.log(msg, "error");
+    }
+
+    async function validateFunction() {
+        let messages = [];
+
+        if (!ncUtil)
+            messages.push("ncUtil was not provided");
+        else if (!channelProfile)
+            messages.push("channelProfile was not provided");
+        else if (!channelProfile.channelSettingsValues)
+            messages.push("channelProfile.channelSettingsValues was not provided");
+        else if (!channelProfile.channelSettingsValues.namespaces)
+            messages.push("channelProfile.channelSettingsValues.protocol was not provided");
+        else if (!channelProfile.channelSettingsValues.wsdl_uri)
+            messages.push("channelProfile.channelSettingsValues.api_uri was not provided");
+        else if (!channelProfile.channelAuthValues)
+            messages.push("channelProfile.channelAuthValues was not provided");
+        else if (!channelProfile.channelAuthValues.account)
+            messages.push("channelProfile.channelAuthValues.account was not provided");
+        else if (!channelProfile.channelAuthValues.consumerKey)
+            messages.push("channelProfile.channelAuthValues.consumerKey was not provided");
+        else if (!channelProfile.channelAuthValues.consumerSecret)
+            messages.push("channelProfile.channelAuthValues.consumerSecret was not provided");
+        else if (!channelProfile.channelAuthValues.tokenID)
+            messages.push("channelProfile.channelAuthValues.tokenID was not provided");
+        else if (!channelProfile.channelAuthValues.tokenSecret)
+            messages.push("channelProfile.channelAuthValues.tokenSecret was not provided");
+        else if (!channelProfile.customerBusinessReferences)
+            messages.push("channelProfile.customerBusinessReferences was not provided");
+        else if (!nc.isArray(channelProfile.customerBusinessReferences))
+            messages.push("channelProfile.customerBusinessReferences is not an array");
+        else if (!nc.isNonEmptyArray(channelProfile.customerBusinessReferences))
+            messages.push("channelProfile.customerBusinessReferences is empty");
+        else if (!payload)
+            messages.push("payload was not provided");
+        else if (!payload.doc)
+            messages.push("payload.doc was not provided");
+
+        if (messages.length > 0) {
+            messages.forEach(msg => logError(msg));
+            out.ncStatusCode = 400;
+            throw new Error(`Invalid request [${messages.join(", ")}]`);
```

Comment:
Since the array will never have more than one message, there's not really anything to loop through or join.